weave/__init__.py

```diff
@@ -71,6 +73,7 @@
     "EvaluationLogger",
     "File",
     "finish",
+    "generate_id",
```

Comment:
Why are we export `generate_id`? Do we expect a user to call this directly?

Reply:
I think we should try to limit the re-exporting of symbols as much as possible. This is primarily intended to surface the most important symbols to avoid deeper imports. I would instead make `weave thread() as t` yield a "Thread" class that has an "id" property containing a generated ID if one is not passed. This would allow the user to store that thread id if needed, but not need to have extra code / imports